backend/src/routes/api/service/modelregistry/index.ts

```diff
@@ -1,32 +1,18 @@
-import httpProxy from '@fastify/http-proxy';
-import { KubeFastifyInstance } from '../../../../types';
-import { DEV_MODE, MODEL_REGISTRY_NAMESPACE } from '../../../../utils/constants';
-import { getParam, setParam } from '../../../../utils/proxy';
+import { MODEL_REGISTRY_NAMESPACE } from '../../../../utils/constants';
+import { proxyService } from '../../../../utils/proxy';
 
-export default async (fastify: KubeFastifyInstance): Promise<void> => {
-  fastify.register(httpProxy, {
-    upstream: '',
-    prefix: '/:name',
-    rewritePrefix: '',
-    replyOptions: {
-      // preHandler must set the `upstream` param
-      getUpstream: (request) => getParam(request, 'upstream'),
-    },
-    preHandler: (request, _, done) => {
-      const name = getParam(request, 'name');
-
-      const upstream = DEV_MODE
-        ? // Use port forwarding for local development:
-          // kubectl port-forward -n <namespace> svc/<service-name> <local.port>:<service.port>
-          `http://${process.env.MODEL_REGISTRY_SERVICE_HOST}:${process.env.MODEL_REGISTRY_SERVICE_PORT}`
-        : // Construct service URL
-          `http://${name}.${MODEL_REGISTRY_NAMESPACE}.svc.cluster.local:8080`;
-
-      // assign the `upstream` param so we can dynamically set the upstream URL for http-proxy
-      setParam(request, 'upstream', upstream);
-
-      fastify.log.info(`Proxy ${request.method} request ${request.url} to ${upstream}`);
-      done();
-    },
-  });
-};
+export default proxyService(
+  null,
+  {
+    port: 8080,
+    namespace: MODEL_REGISTRY_NAMESPACE,
+  },
+  {
+    // Use port forwarding for local development:
+    // kubectl port-forward -n odh-model-registries svc/<service-name> 8085:8080
+    host: process.env.MODEL_REGISTRY_SERVICE_HOST,
+    port: process.env.MODEL_REGISTRY_SERVICE_PORT,
+  },
+  null,
```

Comment:
We didn't implement at first status check for model registry, but it's based on the `ModelRegistry` `CR` availability, in this case we have

```yaml
status:
  conditions:
    - lastTransitionTime: '2024-07-11T19:47:12Z'
      message: Deployment is available
      reason: DeploymentAvailable
      status: 'True'
      type: Available
```
So something like:

```typescript
  (resource) =>
    !!resource.status?.conditions?.find((c) => c.type === 'Available' && c.status === 'True'),
```
would do the trick.

Reply:
+1, this will be the same logic we'll soon use for the Available status in the UI as part of https://issues.redhat.com/browse/RHOAIENG-7485